src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/DoctrineTransport.php

```diff
@@ -37,6 +39,18 @@ public function __construct(Connection $connection, SerializerInterface $seriali
         $this->serializer = $serializer;
     }
 
+    public function __destruct()
+    {
+        // The worker using this transport might have pulled 50 msgs out of the mq, marking them as "in process",
+        // but because of its options (ie --limit, --failure-limit, --memory-limit, --time-limit) it might terminate
```

Comment:
shouldnt you read the options instead :thinking:

Reply:
Then I think we would increase complexity and it wouldn't scale.

For `--limit` we could do some calculations... would be doable despite the increase in complexity..
But for `-failure-limit` we would need to somehow keep track of how many failures happened, probably replicating logic that is already built somewhere. For `--memory-limit` and `--time-limit` the same.

Then, if/when in the future we would add more options, someone would need to also adjust this code.

With this implementation however, we don't need to do anything special to adjust to a particular option, neither now nor in the future if/when we add new options.

Reply:
got ya, i was thinking we might correlate limit and batch_size somehow, but i agree it increases complexity for little gain, and the conceptual issue remains either way

Reply:
im not sure if `__destruct` is the best extension point compared to eg. events, but if it works .. :shrug: